src/rp2_common/pico_cyw43_arch/cyw43_arch.c

```diff
@@ -69,15 +69,19 @@ const char* cyw43_tcpip_link_status_name(int status)
 }
 #endif
 
-int cyw43_arch_wifi_connect_async(const char *ssid, const char *pw, uint32_t auth) {
+
+int cyw43_arch_wifi_connect_bssid_async(const char *ssid, const uint8_t *bssid, const char *pw, uint32_t auth) {
     if (!pw) auth = CYW43_AUTH_OPEN;
     // Connect to wireless
-    return cyw43_wifi_join(&cyw43_state, strlen(ssid), (const uint8_t *)ssid, pw ? strlen(pw) : 0, (const uint8_t *)pw, auth, NULL, CYW43_ITF_STA);
+    return cyw43_wifi_join(&cyw43_state, strlen(ssid), (const uint8_t *)ssid, pw ? strlen(pw) : 0, (const uint8_t *)pw, auth, bssid, CYW43_CHANNEL_NONE);
 }
 
-// Connect to wireless, return with success when an IP address has been assigned
-static int cyw43_arch_wifi_connect_until(const char *ssid, const char *pw, uint32_t auth, absolute_time_t until) {
-    int err = cyw43_arch_wifi_connect_async(ssid, pw, auth);
+int cyw43_arch_wifi_connect_async(const char *ssid, const char *pw, uint32_t auth) {
+    return cyw43_arch_wifi_connect_bssid_async(ssid, NULL, pw, auth);
+}
+
+static int cyw43_arch_wifi_connect_bssid_until(const char *ssid, const uint8_t *bssid, const char *pw, uint32_t auth, absolute_time_t until) {
+    int err = cyw43_arch_wifi_connect_bssid_async(ssid, bssid, pw, auth);
```

Comment:
second invocation of this in the loop does not use `_bssid`

Reply:
well spotted!